src/lines_to_params.rs

```diff
@@ -0,0 +1,165 @@
+use crate::parser;
+use pyo3::prelude::*;
+use pyo3::types::PyDict;
+
+type ActionEffectParams = (u8, u8, u8, u8, u16, u8, u8);
+type StatusEffectParams = (u16, u16, f32, u32);
+
+type StatuslistParams<'a> = (
+    (u32, &'a str),
+    &'a str,
+    Vec<u32>,
+    Vec<f32>,
+    Vec<(u16, u16, f32, u32)>,
+);
+
+/// Get id name type from tuple
+#[pyfunction]
+#[pyo3(text_signature = "(name_id_pair: list[str]) -> (int, str)")]
+pub(crate) fn parse_id_name_pair(inp: Vec<&str>) -> (u32, &str) {
+    (
+        parser::u32_from_param(inp.first().unwrap()),
+        inp.last().unwrap(),
+    )
+}
+
+/// Params to ability
+#[pyfunction]
+#[pyo3(text_signature = "(timestamp: int, params: list[str]) -> Ability")]
+pub(crate) fn ability_from_params(timestamp: i64, inp: Vec<&str>) -> PyObject {
```

Comment:
Does this return an Ability or a PyObject?

Reply:
It returns an Ability but in rust they dont know what the Ability object is and Pyo3 for it to build properly needs it be exported as PyObject

Reply:
There's a proper way to do this. Take a look thru https://pyo3.rs/v0.17.2/class.html and https://pyo3.rs/v0.17.2/conversions/traits.html, maybe also https://pyo3.rs/v0.17.2/python_from_rust.html

AFAIK you can instantiate the actual python Ability class and reference it as a rust object.

Reply:
These are if you want to define a rust class to be exposed to python and not the other way around that I'm using.